### PR TITLE
Toch

### PR DESCRIPTION
**Pull request by Tochukwuibe**
Adds ItemFeedback and LearningItem support

### CODE REVIEW

File: src/main/java/com/google/sps/data/ItemFeedbackService.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    public void store(ItemFeedback feedback) {
+
+        Entity task = new Entity(ITEM_FEEDBACK, feedback.getId());
+
+
+
+        task.setProperty("learningPath", feedback.getLearningPath());
+		task.setProperty("userId", feedback.getUserId());
+        task.setProperty("rating", feedback.getRating());
+        task.setProperty("rating", feedback.getRating());
```
Comment (phaas) [inline src/main/java/com/google/sps/data/ItemFeedbackService.java]: Duplicated statement?

File: src/main/java/com/google/sps/data/ItemFeedbackService.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Query;
+
+public class ItemFeedbackService {
+
+    private static final String ITEM_FEEDBACK = "itemFeedback";
+
+    private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    public List<ItemFeedback> load(long itemId) {
```
Comment (phaas) [inline src/main/java/com/google/sps/data/ItemFeedbackService.java]: itemId should be `learningPathId`. To make this clearer, the method name could be something like `findByLearningPath(..)`.

**Review by phaas — CHANGES_REQUESTED** [review]
In the test, please update `assertSameSection` to check that the item count and contents match.

For the data access patterns:
 - Add a new feedback item for [user, path, item] (done)
 - Update an existing feedback item (done)
 - Find all feedback for a [user, path] so we can compute the completion progress (missing)

File: src/main/java/com/google/sps/data/ItemFeedbackService.java (hunk 1 above)
Comment (phaas) [reply src/main/java/com/google/sps/data/ItemFeedbackService.java]: I think this should be setting "completed"

File: src/main/java/com/google/sps/data/LearningPathService.java [hunk 3]
```diff
@@ -77,20 +77,45 @@ public LearningPath load(long id) throws EntityNotFoundException {
...
+		return sections.stream().map(e -> {
+
+			LearningSection section = new LearningSection(e.getKey().getId(), (String) e.getProperty("name"),
+					"description", (long) e.getProperty("sequence"));
+
+			List<LearningItem> items = loadItems(section.getId());
+
+			section.getItems().addAll(items);
+
+			return section;
```
Comment (phaas) [inline src/main/java/com/google/sps/data/LearningPathService.java]: I think there's enough code here that we should move this to a separate function that takes an Entity and creates a LearningSection.

`return sections.stream().map(this::createSection)....`

File: src/main/java/com/google/sps/data/LearningPathService.java [hunk 4]
```diff
@@ -77,20 +77,45 @@ public LearningPath load(long id) throws EntityNotFoundException {
...
+		}).collect(Collectors.toList());
+	}
+
+	private List<LearningItem> loadItems(long id) {
+		Query query = new Query(LEARNING_ITEM).addSort("sequence")
+				.setFilter(new Query.FilterPredicate("learningSection", Query.FilterOperator.EQUAL, id));
+
+		List<Entity> sections = datastore.prepare(query).asList(FetchOptions.Builder.withDefaults());
```
Comment (phaas) [inline src/main/java/com/google/sps/data/LearningPathService.java]: `sections` should be `items` now

**Comment by Tochukwuibe** [comment]
Changes made.

**Review by phaas — APPROVED** [review]
Looks good to me!

**Pull request merged**